### ISSUE TITLE
Crash with Gothic II (Gold Edition)

### ISSUE THREAD

**Issue by kapitaenzufall** [Try/OpenGothic#20]
Hi!
I tried to run the pre-release gothic-win32-v1.0.69 with a new Gothic II (Gold Edition) installation from DVD on Windows 10 but got a crash.

Executed:
> .\Gothic2Notr.exe -g "D:\JoWooD\Gothic II"

Crash log:
> unable to read .ini file

> ---crashlog(SIGABRT)---
0x00470abc: CrashLog::dumpStack(char const*) in d:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x00470989: signalHandler(int) in d:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x75e972b1: raise in C:\WINDOWS\System32\msvcrt.dll
0x75e9b76b: abort in C:\WINDOWS\System32\msvcrt.dll
0x004709ae: CrashLog::setup() in d:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x6fe59399: ZN10__cxxabiv111__terminateEPFvvE in d:\JoWooD\open_gothic\bin\libstdc++-6.dll
0x0045376a: Renderer::initSwapchain(unsigned int, unsigned int) in d:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x0045dd04: MainWindow::initSwapchain() in d:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x0045b9f4: MainWindow::resizeEvent(Tempest::SizeEvent&) in d:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x6b6b99f4: ZN7Tempest6Widget6resizeEii in d:\JoWooD\open_gothic\bin\libMoltenTempest.dll
0x6b6d7a09: ZN7Tempest6Window4Impl8onResizeEPNS_9SystemApi6WindowEii in d:\JoWooD\open_gothic\bin\libMoltenTempest.dll
0x6b684e95: ZN7Tempest9SystemApi10showCursorEb in d:\JoWooD\open_gothic\bin\libMoltenTempest.dll
0x760d48eb: AddClipboardFormatListener in C:\WINDOWS\System32\USER32.dll
0x760b613c: CallWindowProcW in C:\WINDOWS\System32\USER32.dll
0x760b568f: CallWindowProcW in C:\WINDOWS\System32\USER32.dll
0x6edfea7e: glDebugEntry in C:\WINDOWS\SYSTEM32\opengl32.dll
0x760d48eb: AddClipboardFormatListener in C:\WINDOWS\System32\USER32.dll
0x760b613c: CallWindowProcW in C:\WINDOWS\System32\USER32.dll
0x760b5cda: CallWindowProcW in C:\WINDOWS\System32\USER32.dll
0x760be6bf: IsRectEmpty in C:\WINDOWS\System32\USER32.dll
0x77c740ed: KiUserCallbackDispatcher in C:\WINDOWS\SYSTEM32\ntdll.dll
0x760b799d: GetSystemMetrics in C:\WINDOWS\System32\USER32.dll
0x760cac40: UnionRect in C:\WINDOWS\System32\USER32.dll
0x73b198eb: OpenThemeData in C:\WINDOWS\system32\uxtheme.dll
0x73b27c18: GetThemeBackgroundRegion in C:\WINDOWS\system32\uxtheme.dll
0x760c6ba7: DefMDIChildProcA in C:\WINDOWS\System32\USER32.dll
0x6b684ed3: ZN7Tempest9SystemApi10showCursorEb in d:\JoWooD\open_gothic\bin\libMoltenTempest.dll
0x760d48eb: AddClipboardFormatListener in C:\WINDOWS\System32\USER32.dll
0x760b613c: CallWindowProcW in C:\WINDOWS\System32\USER32.dll
0x760b568f: CallWindowProcW in C:\WINDOWS\System32\USER32.dll
0x6edfea7e: glDebugEntry in C:\WINDOWS\SYSTEM32\opengl32.dll
0x760d48eb: AddClipboardFormatListener in C:\WINDOWS\System32\USER32.dll
0x760b613c: CallWindowProcW in C:\WINDOWS\System32\USER32.dll
0x760b5cda: CallWindowProcW in C:\WINDOWS\System32\USER32.dll
0x760beed7: RemovePropW in C:\WINDOWS\System32\USER32.dll
0x77c740ed: KiUserCallbackDispatcher in C:\WINDOWS\SYSTEM32\ntdll.dll
0x0045dad8: MainWindow::setFullscreen(bool) in d:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x0045a644: MainWindow::MainWindow(Gothic&, Tempest::VulkanApi&) in d:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x0045a0a8: _main in d:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x004013e2: ___tmainCRTStartup in d:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x75716359: BaseThreadInitThunk in C:\WINDOWS\System32\KERNEL32.DLL
0x77c67a94: RtlGetAppContainerNamedObjectPath in C:\WINDOWS\SYSTEM32\ntdll.dll
0x77c67a64: RtlGetAppContainerNamedObjectPath in C:\WINDOWS\SYSTEM32\ntdll.dll

**Comment by Try** [comment]
Hmm, its quite weird to see crash in:
```
0x6fe59399: std::terminate
0x0045376a: Renderer::initSwapchain
```
Can you please test how it's works in window mode? 
> .\Gothic2Notr.exe -window -g "D:\JoWooD\Gothic II"

**Comment by kapitaenzufall** [comment]
In window mode I got as crash log:

> ---crashlog(SIGABRT)---
0x00470abc: CrashLog::dumpStack(char const*) in D:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x00470989: signalHandler(int) in D:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x75e972b1: raise in C:\WINDOWS\System32\msvcrt.dll
0x75e9b76b: abort in C:\WINDOWS\System32\msvcrt.dll
0x004709ae: CrashLog::setup() in D:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x6fe59399: ZN10__cxxabiv111__terminateEPFvvE in D:\JoWooD\open_gothic\bin\libstdc++-6.dll
0x0045376a: Renderer::initSwapchain(unsigned int, unsigned int) in D:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x0045dd04: MainWindow::initSwapchain() in D:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x0045a6b0: MainWindow::MainWindow(Gothic&, Tempest::VulkanApi&) in D:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x0045a0a8: _main in D:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x004013e2: ___tmainCRTStartup in D:\JoWooD\open_gothic\bin\Gothic2Notr.exe
0x75716359: BaseThreadInitThunk in C:\WINDOWS\System32\KERNEL32.DLL
0x77c67a94: RtlGetAppContainerNamedObjectPath in C:\WINDOWS\SYSTEM32\ntdll.dll
0x77c67a64: RtlGetAppContainerNamedObjectPath in C:\WINDOWS\SYSTEM32\ntdll.dll

Whats about the ini file that cant be found? This line gets printed after the executed command but not in the crash log:

> unable to read .ini file

**Comment by Try** [comment]
> Whats about the ini file that cant be found? 
No worry about it: basically game is yelling about no Gothic.ini in working directory.

About the crash itself: the problem is in line 39 of renderer.cpp:
```
  zbuffer        = device.createTexture(TextureFormat::Depth24x8,w,h,false);
`` 
I always expect D24x8 to be available. What graphics card do you have?

**Comment by Try** [comment]
> What graphics card do you have?
Anyway I was able to reproduce it on intel-hd-graphics-530.

**Comment by kapitaenzufall** [comment]
> What graphics card do you have?

NVIDIA Geforce GTX 1060 6GB

> Anyway I was able to reproduce it on intel-hd-graphics-530.

Great! I'll test it once there is a new pre-release.

**Comment by kapitaenzufall** [comment]
Fixed with gothic-win32-v1.0.115 :)
